### DISCUSSION TITLE
Check for and Attempt to Fetch a High-Res DEM : Using 3DEP LidAR DEM or OpenTopography APIs

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/QRiS#113] [Ideas & Feature Requests]
@philipbaileynar after our relative success (noting need for expectation management) in adding the StreamStats API access, I am wondering if we dare explore another one. One of the most common basemaps someone wants to get for their locality is a LIDAR DEM. We know what we would like to do with that DEM. We have success in RS_Context in pining their API and getting DEMs.  Revisiting https://github.com/Riverscapes/riverscapes-tools/issues/145 inspired me a bit on this one. 

This would work as follows:
- `Check for Available High Resolution Topography` - This would be available command within project for Basemaps, or outside of project in the Non-Project Commands menu.  It would prompt user to specify a polygon (e.g. from mask library) and then it would by default have checked to look on 3DEP and on OpenTopography (see [OT API documentation here](https://opentopography.org/developers)). It would then return option to download and clip to polygon AOI. We would have all sorts of warnings directing complaints about this to the provider of the API and deflecting problems away from ourselves. 
- If you clicked on `Download and Clip to AOI` It would basically return a GeoTIFF (just like RS_Context does) with some obvious functional limits (i.e. can't ask for an area too large) and then if you did it inside a project, we invoke our ever favorite "Hillshade Giggle Moment" - i.e. symbolize, add to map transparently and derive hillshade and add it right beneath it. 

I predict this stupid tool (if we tracked how often different services go used), would be one of our most used commands and create loyal customers for life. As more DEMs available through APIs came on line, we could 

-----
**Disclaimer** - I am fine if this gets added to the stack for after 1.X Release. I just want to see if there are any similarly useful or easy ones we can add.

**Comment by joewheaton** [comment]
Look how much more complete this has gotten on OT:
![image](https://user-images.githubusercontent.com/9723905/192152760-e070c3d3-2174-4bfb-a532-bd1f2f7b05fe.png)
This might even be a little inspiration for showing multiple projects in Warehouse Map discovery for Tyler:
![image](https://user-images.githubusercontent.com/9723905/192152863-430fd2ca-461f-4685-bd04-1ae598927433.png)

**Reply by philipbaileynar** [reply]
I am really energized for this, primarily because it is a stepping stone to Riverscapes Studio.

FWIW last time we explored 3DEP there was no ability to "check for presence" of data... you just requested data for an extent polygon at a certain resolution and it returned data or it didn't.

**Comment by joewheaton** [comment]
@philipbaileynar, once 0.5 is released and @jtgilbert is done with some current firedrills, lets get you, @KellyMWhitehead and @jtgilbert together to crack this. We have a bunch of projects where this is going to be useful in Anabranch. I can pay for this now and I think this will be something that goes in store (above certain sizes) and we should have in 1.0.  Thoughts?
